multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/UpdateServiceMetadataExecution.java

```diff
@@ -0,0 +1,37 @@
+package org.cloudfoundry.multiapps.controller.process.steps;
+
+import java.text.MessageFormat;
+
+import org.cloudfoundry.client.lib.CloudControllerClient;
+import org.cloudfoundry.client.lib.domain.CloudMetadata;
+import org.cloudfoundry.client.lib.domain.CloudServiceInstance;
+import org.cloudfoundry.client.lib.domain.ImmutableCloudServiceInstance;
+import org.cloudfoundry.multiapps.controller.client.lib.domain.CloudServiceInstanceExtended;
+import org.cloudfoundry.multiapps.controller.process.variables.Variables;
+
+public class UpdateServiceMetadataExecution implements AsyncExecution {
+
+    @Override
+    public AsyncExecutionState execute(ProcessContext context) {
+        CloudServiceInstanceExtended serviceInstance = context.getVariable(Variables.SERVICE_TO_PROCESS);
+        updateMetadata(context.getControllerClient(), serviceInstance);
+        return AsyncExecutionState.FINISHED;
+    }
+
+    private void updateMetadata(CloudControllerClient client, CloudServiceInstanceExtended serviceInstance) {
+        CloudMetadata serviceMetadata = client.getServiceInstance(serviceInstance.getName())
```

Comment:
Don't get this metadata confused with the V3 metadata. This one is just the V2 metadata of the resource that includes its GUID, created/updated timestamps, etc. The V3 metadata comes from the `getV3Metadata()` method.